Library/Homebrew/utils/curl.rb

```diff
@@ -3,9 +3,10 @@
 
 def curl_executable
   curl = Pathname.new ENV["HOMEBREW_CURL"]
-  curl = which("curl") unless curl.exist?
-  return curl if curl.executable?
-  raise "#{curl} is not executable"
+  curl = which("curl") unless curl.executable?
+  curl = "/usr/bin/curl" unless curl.executable?
```

Comment:
I find the redefinition followed by a conditional checking the previous value quite confusing.

What would you think about something like this?

```ruby
curl = [which("curl"), "/usr/bin/curl"].find(&:executable?)
raise "/usr/bin/curl is not executable" if curl.nil?
```

Reply:
(Note that if the exception is raised, it will, AFAICT, _always_ say /usr/bin/curl, which isn't obvious from the code as it stands)

Reply:
@alyssais Thanks, yeh, that's much nicer. Have tweaked this further, let me know what you think.